### PR TITLE
Add binary variant to PUBSUB subscribe commands

### PR DESCRIPTION
**Pull request by yipin-chen**
Add binary variant for the following commands:
- `PSubbscribe`
- `Subscribe`
- `SSubscribe`
- `Unsubscribe`
- `SUnsubscribe`
- `PUnsubscribe`

Add decoder option to PUBSUB's `createClient()`.

Note: to complete `PUBSUB` subscribe commands, I have to cherry-pick already merged main branch PR #2201.

### CODE REVIEW

File: node/src/BaseClient.ts [hunk 1]
```diff
@@ -957,13 +957,13 @@ export class BaseClient {
                 nextPushNotificationValue = valueFromSplitPointer(
                     responsePointer.high,
                     responsePointer.low,
-                    decoder === Decoder.String,
+                    (decoder ?? this.defaultDecoder) === Decoder.String,
```
Comment (acarbonetto) [inline node/src/BaseClient.ts:960]: common code, we can save this to a variable and reduce code duplication

File: node/tests/PubSub.test.ts [hunk 2]
```diff
@@ -92,14 +93,17 @@ describe("PubSub", () => {
...
             | GlideClusterClientConfiguration.PubSubSubscriptions,
+        decoder?: Decoder,
     ): Promise<[TGlideClient, TGlideClient]> {
         let client: TGlideClient | undefined;
 
         if (clusterMode) {
             try {
                 options.pubsubSubscriptions = pubsubSubscriptions;
+                options.defaultDecoder = decoder;
                 client = await GlideClusterClient.createClient(options);
```
Comment (acarbonetto) [inline node/tests/PubSub.test.ts:104]: Instead of editing the input argument object.
```suggestion
                client = await GlideClusterClient.createClient({
                    pubsubSubscriptions = pubsubSubscriptions,
                    defaultDecoder = decoder,
                    ...options
                });
```

File: node/tests/PubSub.test.ts [hunk 3]
```diff
@@ -92,14 +93,17 @@ describe("PubSub", () => {
...
                 options2.pubsubSubscriptions = pubsubSubscriptions2;
+                options2.defaultDecoder = decoder;
                 const client2 = await GlideClusterClient.createClient(options2);
```
Comment (acarbonetto) [inline node/tests/PubSub.test.ts:107]: same here

File: node/tests/PubSub.test.ts [hunk 4]
```diff
@@ -112,8 +116,10 @@ describe("PubSub", () => {
         } else {
             try {
                 options.pubsubSubscriptions = pubsubSubscriptions;
+                options.defaultDecoder = decoder;
                 client = await GlideClient.createClient(options);
```
Comment (acarbonetto) [inline node/tests/PubSub.test.ts:120]: and here

File: node/tests/PubSub.test.ts [hunk 5]
```diff
@@ -112,8 +116,10 @@ describe("PubSub", () => {
...
                 options2.pubsubSubscriptions = pubsubSubscriptions2;
+                options2.defaultDecoder = decoder;
                 const client2 = await GlideClient.createClient(options2);
```
Comment (acarbonetto) [inline node/tests/PubSub.test.ts:123]: and here

File: node/tests/PubSub.test.ts [hunk 6]
```diff
@@ -387,9 +371,96 @@ describe("PubSub", () => {
...
+                const pubsubMessage = await getMessageByMethod(
+                    method,
+                    listeningClient,
+                    context,
+                    0,
+                );
+
+                expect(pubsubMessage!.message).toEqual(Buffer.from(message));
+                expect(pubsubMessage!.channel).toEqual(Buffer.from(channel));
+                expect(pubsubMessage!.pattern).toEqual(null);
```
Comment (acarbonetto) [inline node/tests/PubSub.test.ts:463]: there's `toBeNull()`

File: node/tests/PubSub.test.ts [hunk 7]
```diff
@@ -387,9 +371,96 @@ describe("PubSub", () => {
                     context,
                     0,
                 );
-                expect(pubsubMessage.message).toEqual(message);
-                expect(pubsubMessage.channel).toEqual(channel);
-                expect(pubsubMessage.pattern).toEqual(null);
+
+                expect(pubsubMessage!.message).toEqual(message);
+                expect(pubsubMessage!.channel).toEqual(channel);
+                expect(pubsubMessage!.pattern).toEqual(null);
```
Comment (acarbonetto) [inline node/tests/PubSub.test.ts:377]: there's `toBeNull()`

File: node/tests/PubSub.test.ts [hunk 8]
```diff
@@ -570,25 +640,21 @@ describe("PubSub", () => {
...
                 // Check if all messages are received correctly
                 for (let index = 0; index < NUM_CHANNELS; index++) {
-                    const pubsubMsg = await getMessageByMethod(
+                    const pubsubMsg = (await getMessageByMethod(
                         method,
                         listeningClient,
                         context,
                         index,
-                    );
+                    ))!;
```
Comment (acarbonetto) [inline node/tests/PubSub.test.ts:648]: why did you add this?

**Review by acarbonetto — APPROVED** [review]
lgtm

File: node/tests/PubSub.test.ts (hunk 2 above)
Comment (yipin-chen) [reply node/tests/PubSub.test.ts:104]: I have tried your suggested implementation but the test failed.

File: node/tests/PubSub.test.ts (hunk 8 above)
Comment (yipin-chen) [reply node/tests/PubSub.test.ts:648]: VS Code complaints that `pubsubMsg` is possibly `null`. So I have to add this.

File: node/tests/PubSub.test.ts (hunk 2 above)
Comment (yipin-chen) [reply node/tests/PubSub.test.ts:104]: I have resolved the issues.

File: node/tests/PubSub.test.ts
Comment (Yury-Fridlyand) [inline node/tests/PubSub.test.ts:1]: I don't review tests, I trust you. If they past - all LGTM

**Review by Yury-Fridlyand — APPROVED** [review]

File: node/tests/PubSub.test.ts
Comment (yipin-chen) [reply node/tests/PubSub.test.ts:1]: All PubSub tests passed.
